src/emulator/databaseEmulator.ts

```diff
@@ -5,6 +5,8 @@ import { Constants } from "./constants";
 interface DatabaseEmulatorArgs {
   port?: number;
   host?: string;
+  functions_emulator_port?: number;
+  functions_emulator_host?: string;
```

Comment:
Is there a reason these are snake_case? Prefer lowerCamelCase

Reply:
They get translated directly into command-line args for the JAR:
```
javaEmulators.start(Emulators.DATABASE, this.args);
```

But I guess we should add a comment making this more explicit!  Or a transformation function so that they can be `camelCase` in the interface def but `snake_case` when used.

Reply:
In an offline discussion, @yuchenshi and I settled on consistency with the firestore emulator: https://github.com/firebase/firebase-tools/blob/master/src/emulator/firestoreEmulator.ts#L10. I would prefer to use snake case since we've already merged the RTDB change that parses these arguments.

Reply:
The same comment also applies to `firestoreEmulator.ts` and I think we should add the same comment / transformation function there as well. My vote is to merge this in and then do a follow-up PR to add the transformation function in `javaEmulators.ts`.

Reply:
Yeah the transformation function would be generally useful for more than just casing, I recently ran into an issue in another PR of mine:
https://github.com/firebase/firebase-tools/pull/1358/files#diff-b7bbe6f5b9ebb932f1d211213eb8874d

We will want to be able to provide args to the `FooEmulator` classes that don't necessarily get passed directly into the JAR.  So the function could do casing as well as whitelisting.

Reply:
Yeah, I would come back and clean these up for sure. General style guide should take precedence here; it's not hard to re-create the args (and it's _way, way_ more explicit anyway, which is good), so that's the path that should be taken.

Reply:
I'm happy to submit a follow-up PR that performs a transformation and applies a whitelist before building the command: https://github.com/firebase/firebase-tools/blob/master/src/serve/javaEmulators.ts#L75-L80.